### PR TITLE
Bare-metal Form and helper reformer

### PR DESCRIPTION
**Pull request by mihrab34**
Rails form project focused on how to use various rails form helpers for form creation and display data received

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email],password: params[:password])
+    @user = User.new(user_params)
```
Comment (trekab) [inline app/controllers/users_controller.rb]: Comments: It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  validates :username, presence: true
+  validates :email, presence: true
+  # format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i }
+  validates :password, presence: true
```
Comment (trekab) [inline app/models/user.rb]: Comments: It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

**Review by trekab — CHANGES_REQUESTED** [review]
## Changes Required! :recycle: 
Hi @mihrab34! Well done following this project's guidelines so far! I commend you for working with a form partial and displaying a list of error messages in your views! :+1:  There are some changes I'd like you to look at before you can have this project approved. These include:
- [ ] Make sure that you use config files copied from [config linter repo](https://github.com/microverseinc/linters-config#linters-config). I have noticed that you have set up both `stickler` and `GitHub Actions` for your project. The `linters.yml` file you used is for the ruby section. Just follow the instructions for setting up `GitHub Actions` as indicated in [this](https://github.com/microverseinc/linters-config/tree/master/ror) repo's README file.
- [ ] You need to implement an `update` action in order to be able to update user information. This action needs to have the following properties/details;
  - it should find the user based on id from params, i.e. `@user = User.find(params[:id])` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_params.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
  - it should `redirect_to edit_user_path` if the user was saved
  - it should `render :edit` if user was not saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_update.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
- [ ] __[OPTIONAL]__ It would be nice if you validated the email format using a regular expression.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[GitHub](https://github.com/trekab) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

**Review by MarioDena — CHANGES_REQUESTED** [review]
Hello Great work on your project 👍 however there is a basic mistake you should be always aware not to do:

## Changes required 

- [ ] On your commits after the review you have repeated 7 times the same commit message "Linter error fixed" Remember all your commit messages should be meaningful this repetition makes your repositories look unprofessional. Remember this is mandatory for all projects. Please either rebase to edit or merge the repeated comments.

## Optional changes

- [ ] I would also recommend making mention of the review changes you implemented when applicable.

When you are done feel free to resubmit for another code review. 

Happy coding!! 🥇

**Review by rOluochKe — APPROVED** [review]
Hello Team!
### Well done, you got this right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next project.
- Happy coding!

**Pull request merged**